src/tty-ask-password-agent/tty-ask-password-agent.c

```diff
@@ -525,7 +525,10 @@ static int watch_passwords(void) {
                 return log_error_errno(errno, "Failed to allocate directory watch: %m");
 
         if (inotify_add_watch(notify, "/run/systemd/ask-password", IN_CLOSE_WRITE|IN_MOVED_TO) < 0)
-                return log_error_errno(errno, "Failed to add /run/systemd/ask-password to directory watch: %m");
+                if (errno == ENOSPC)
+                        return log_error("Failed to add /run/systemd/ask-password to directory watch: inotify watch limit reached");
+                else
+                        return log_error_errno(errno, "Failed to add /run/systemd/ask-password to directory watch: %m");
```

Comment:
please enclose the inner if/else in {}, otherwise it's not obvious to which `if` the `else`you added belongs. actually the compiler should have warned you about that already? it used to do that at least...